### PR TITLE
Oxfordshire WDM integration

### PR DESCRIPTION
**Pull request by struan**
Adds integration with WDM for OCC.

Only handles new reports, new updates and fetching updates. OCC categories are configured on FMS for the time being.

### CODE REVIEW

File: perllib/Open311/Endpoint/Integration/WDM.pm [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package Open311::Endpoint::Integration::WDM;
+
+use Moo;
+extends 'Open311::Endpoint';
+with 'Open311::Endpoint::Role::mySociety';
+
+use DateTime::Format::Strptime;
+
+use Integrations::WDM;
+use Open311::Endpoint::Service::UKCouncil::Oxfordshire;
```
Comment (davea) [inline perllib/Open311/Endpoint/Integration/WDM.pm]: This creates a coupling between WDM and Oxfordshire-specific stuff. Will that cause issues if/when another WDM integration comes along?

File: perllib/Open311/Endpoint/Integration/WDM.pm [hunk 2]
```diff
@@ -42,6 +53,33 @@ sub post_service_request {
...
+        # ignore non FMS references
+        next unless $service_request_id =~ /^\d+$/;
+        my $customer_reference = $update->{ENQUIRY_REFERENCE};
+        $customer_reference =~ s/\s*$//;
+        push @updates, Open311::Endpoint::Service::Request::Update::mySociety->new(
+            status => lc $update->{STATUS},
+            update_id => $update->{UpdateID},
+            service_request_id => $service_request_id,
+            customer_reference => $customer_reference,
+            description => $update->{COMMENTS},
```
Comment (davea) [inline perllib/Open311/Endpoint/Integration/WDM.pm]: In the Confirm integration we have a `publish_service_update_text` flag which must be set in order for the comment text to be published. Worth doing something similar here, or have WDM users been trained about what will/won't be published?

File: perllib/Integrations/WDM.pm [hunk 3]
```diff
@@ -118,6 +118,43 @@ sub post_request {
...
+            }
+        }
+    };
+    my $response = $self->soap_post($self->updates_endpoint, 'UpdateWdmEnquiry', 'xDoc', $self->_create_xml_string($data));
+
+    my $resp_text = $response->valueof('//UpdateWdmEnquiryResponse/UpdateWdmEnquiryResult');
+    if ($resp_text eq 'OK') {
+        return {
+            status => $args->{status},
+            # XXX I am not sure this is correct
```
Comment (davea) [inline perllib/Integrations/WDM.pm]: Was this resolved, can the comment be removed/updated?

File: perllib/Open311/Endpoint/Integration/WDM.pm (hunk 2 above)
Comment (struan) [reply perllib/Open311/Endpoint/Integration/WDM.pm]: The latter is the plan.

File: perllib/Open311/Endpoint/Integration/WDM.pm (hunk 1 above)
Comment (struan) [reply perllib/Open311/Endpoint/Integration/WDM.pm]: I think we'll probably need to see when that happens. It's hard to tell at the moment what's generic about this and what's custom.

**Pull request merged**